scipy/interpolate/tests/test_interpolate.py

```diff
@@ -906,6 +906,24 @@ def test_eval_3d(self):
         yi[:, 1, 1] = 4. * yi_
         xp_assert_close(ak(xi), yi)
 
+    def test_linear_interpolant_edge_case(self):
+        # Reference - https://github.com/scipy/scipy/issues/22011
+        x = np.array([0.0, 1.0], dtype=float)
+        y = np.array([0.55, 0.55])
+        akima = Akima1DInterpolator(x, y, axis=0, extrapolate=None)
+        xp_assert_close(akima(0.45), np.array(0.55))
```

Comment:
Nit: could you update this test to use a non-zero slope. Ideally it should be trivial to check that the result value is correct from just eyeballing the test (it is here, which is good; but the test is weaker than it could be with both `y` values being the same)